api/envoy/extensions/filters/http/cache/v3/cache.proto

```diff
@@ -49,9 +50,13 @@ message CacheConfig {
     repeated config.route.v3.QueryParameterMatcher query_parameters_excluded = 4;
   }
 
-  // Config specific to the cache storage implementation.
+  // Config specific to the cache storage implementation. Required unless ``disabled``
+  // is true.
   // [#extension-category: envoy.http.cache]
-  google.protobuf.Any typed_config = 1 [(validate.rules).any = {required: true}];
+  google.protobuf.Any typed_config = 1;
+
+  // When true, the cache filter is a no-op filter.
```

Comment:
Again, doing that would probably be more confusing than helpful at this point because overriding by route isn't actually implemented yet. Describing things you can't actually do in the documentation seems worse than nothing.

At this version, the use for supporting disabling (until route overrides are available) is allowing ECDS to turn the filter off.

Which is also why the route-level generic disable isn't useful for this case - you can't toggle that with ECDS, you'd have to update some larger entity (the listener?).

The generic disable also doesn't allow for providing a disabled config at the base level and overriding it to enabled per-route.